app/src/main/java/com/nickytoolchick/agraph/DatasetOptionsActivity.kt

```diff
@@ -0,0 +1,143 @@
+        updateFromEditTexts()
+        updateFromSpinner()
+        updateFromCheckedTV()
+        updateFromPointsList()
+    }
+
+    private fun updateFromEditTexts() {
+        datasetOptions.strokeSize = binding.strokeSizeEditText.text.toString().toFloat()
+        datasetOptions.pointRadius = binding.pointRadiusEditText.text.toString().toFloat()
+    }
+
+    private fun updateFromSpinner() {
+        datasetOptions.color = binding.colorSpinner.selectedItem.toString().toInt()
+    }
+
+    private fun updateFromCheckedTV() {
+        datasetOptions.isSmooth = binding.lineSmoothCheckedTV.isChecked
+    }
+
+    private fun updateFromPointsList() {
+        datasetOptions.points = points.toTypedArray()
+    }
+
+    private fun loadDatasetOptions() {
+        datasetOptions = Json.decodeFromString(mainActivityIntent.getStringExtra("stableDatasetOptions")!!)
+        loadSpinner()
+        loadEditTexts()
+        loadSmoothnessCheckedTV()
+        loadPoints()
+        handleSmoothnessCheckedTVClick()
+        updatePointsTextView()
+    }
+
+    private fun handleSmoothnessCheckedTVClick() {
+        binding.lineSmoothCheckedTV.setOnClickListener {
+            binding.lineSmoothCheckedTV.isChecked = !binding.lineSmoothCheckedTV.isChecked
+        }
+    }
+
+    private fun loadSpinner() {
+        val arrayAdapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, colors)
+        binding.colorSpinner.adapter = arrayAdapter
+        binding.colorSpinner.setSelection(datasetOptions.color)
+    }
+
+    private fun loadEditTexts() {
+        binding.strokeSizeEditText.setText(datasetOptions.strokeSize.toString())
+        binding.pointRadiusEditText.setText(datasetOptions.pointRadius.toString())
+    }
+
+    private fun loadSmoothnessCheckedTV() {
+        binding.lineSmoothCheckedTV.isChecked = datasetOptions.isSmooth
+    }
+
+    private fun loadPoints() {
+        points = datasetOptions.points.toMutableList()
+        updatePointsTextView()
+    }
+
+    private fun addPoint() {
+        val x = binding.newPointXEditText.text.toString().toFloat()
+        val y = binding.newPointYEditText.text.toString().toFloat()
+        val newPoint = Pair(x, y)
+        if (!(points.filter { it == newPoint }.any())) {
+            points.add(newPoint)
+            updatePointsTextView()
+        } else {
+            val toast = Toast.makeText(this, "This point already exists!", Toast.LENGTH_LONG)
+            toast.show()
+        }
+    }
+
+    private fun deletePoint() {
+        val x = binding.newPointXEditText.text.toString().toFloat()
+        val y = binding.newPointYEditText.text.toString().toFloat()
+        val pointToDelete = Pair(x, y)
+        if (points.filter { it == pointToDelete }.any()) {
+            points.remove(pointToDelete)
+            updatePointsTextView()
+        } else {
+            val toast = Toast.makeText(this, "This point does not exist!", Toast.LENGTH_LONG)
+            toast.show()
```

Comment:
You can make a shorter wrapper for this call:
```kotlin
fun showToast(context: Conxet, message: String){
val toast = Toast.makeText(context, message, Toast.LENGTH_LONG)
 toast.show()
}
```
It can be an extension for Toast(), private fuction in this class or separate function in utils object